include/openmc/shared_array.h

```diff
@@ -106,7 +104,17 @@ class SharedArray {
   //! where the internal size of the array needs to be manually updated.
   //
   //! \param size The new size of the container
-  void resize(int64_t size) {size_ = size;}
+  void resize(int64_t size)
+  {
+    if (size > capacity_) {
```

Comment:
True. I already wrote this for my CUDA code since I use an extended vector implementation that supports thread_safe_append on both host and device, then typedef SharedArray as my vector implementation. If you'd prefer removal here, that's fine.

Reply:
Good thinking - that makes a lot of sense on GPU to extend the OpenMC vector type to support the `thread_safe_append()` functionality and just totally replace the SharedArray. For the CPU version though, I think we should remove this logic since it will not currently used, and there's a downside to adding unused/untestable code. I do think it would be good to add a check though or at least a comment somewhere in shared_array.h making clear the idea that `resize()` does not allow for specifying size > capacity.

Reply:
new commit removes this functionality and does an `openmc::fatal_error` if attempting to resize beyond capacity.